File: mikan.go
package main
import "fmt"

var n int

func max(a int, b int) int {
	if a > b {return a}
	return b
}

func mikan(w int) int {
	return max((w + n / 2) / n, 1) * n
}

func main() {
	fmt.Scan(&n)
	var m int
	fmt.Scan(&m)
	for i := 0; i < m; i++ {
		var w int
		fmt.Scan(&w)
		fmt.Println(mikan(w))
	}
}

File: mikan.cs
using System;

public class Program
{
	private static int n;

	private static int Mikan(int w)
	{
		return Math.Max((w + n / 2) / n, 1) * n;
	}

	public static void Main()
	{
		int[] nm = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
		n = nm[0];
		int m = nm[1];
		while (m-- > 0) Console.WriteLine(Mikan(int.Parse(Console.ReadLine())));
	}
}
